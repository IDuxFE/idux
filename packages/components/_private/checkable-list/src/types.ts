/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { VirtualScrollToFn } from '@idux/cdk/scroll'
import type { ExtractInnerPropTypes, ExtractPublicPropTypes, MaybeArray, VKey } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes, PropType } from 'vue'

export interface CheckableListData {
  key?: VKey
  label?: string
  disabled?: boolean
  [key: string]: unknown
}

export interface CheckableListScroll {
  height?: string | number
  fullHeight?: boolean
}

export const checkableListProps = {
  dataSource: Array as PropType<CheckableListData[]>,
  checkable: {
    type: Boolean,
    default: true,
  },
  removable: {
    type: Boolean,
    default: false,
  },
  checked: Function as PropType<(item: CheckableListData) => boolean>,
  customAdditional: { type: Function as PropType<CheckableListCustomAdditional>, default: undefined },
  disabled: Function as PropType<(item: CheckableListData) => boolean>,
  getKey: Function as PropType<(item: CheckableListData) => VKey>,
  labelKey: String,
  virtual: {
    type: Boolean,
    default: false,
  },
  scroll: Object as PropType<CheckableListScroll>,
  onCheckChange: [Function, Array] as PropType<MaybeArray<(item: CheckableListData, checked: boolean) => void>>,
  onRemove: [Function, Array] as PropType<MaybeArray<(item: CheckableListData) => void>>,
  onScroll: [Function, Array] as PropType<MaybeArray<(evt: Event) => void>>,
  onScrolledChange: [Function, Array] as PropType<
    MaybeArray<(startIndex: number, endIndex: number, visibleData: unknown[]) => void>
  >,
  onScrolledBottom: [Function, Array] as PropType<MaybeArray<() => void>>,
} as const

export const checkableListItemProps = {
  checked: {
    type: Boolean,
    default: false,
  },
  checkable: {
    type: Boolean,
    default: true,
  },
  removable: {
    type: Boolean,
    default: false,
  },
  disabled: {
    type: Boolean,
    default: false,
  },
  label: String,
  value: {
    type: [String, Number, Symbol] as PropType<VKey>,
    required: true,
  },
  onCheckChange: [Function, Array] as PropType<MaybeArray<(checked: boolean) => void>>,
  onRemove: [Function, Array] as PropType<MaybeArray<() => void>>,
} as const

export interface CheckableListApi {
  scrollTo: VirtualScrollToFn
}

export type CheckableListProps = ExtractInnerPropTypes<typeof checkableListProps>
export type CheckableListItemProps = ExtractInnerPropTypes<typeof checkableListItemProps>
export type CheckableListPublicProps = ExtractPublicPropTypes<typeof checkableListProps>
export type CheckableListComponent = DefineComponent<
  Omit<HTMLAttributes, keyof CheckableListPublicProps> & CheckableListPublicProps,
  CheckableListApi
>
export type CheckableListInstance = InstanceType<DefineComponent<CheckableListProps, CheckableListApi>>

export type CheckableListCustomAdditional = (options: {
  data: CheckableListData
  index: number
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
}) => Record<string, any> | undefined
