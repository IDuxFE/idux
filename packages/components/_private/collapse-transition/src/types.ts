/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes } from 'vue'

import { IxPropTypes } from '@idux/cdk/utils'

export type CollapseTransitionMode = 'height' | 'width'

export const collapseTransitionProps = {
  appear: IxPropTypes.bool.def(false),
  name: IxPropTypes.string,
  mode: IxPropTypes.oneOf<CollapseTransitionMode>(['height', 'width']).def('height'),
  onAfterEnter: IxPropTypes.emit(),
  onAfterLeave: IxPropTypes.emit(),
}

export type CollapseTransitionProps = ExtractInnerPropTypes<typeof collapseTransitionProps>
export type CollapseTransitionPublicProps = ExtractPublicPropTypes<typeof collapseTransitionProps>
export type CollapseTransitionComponent = DefineComponent<
  Omit<HTMLAttributes, keyof CollapseTransitionPublicProps> & CollapseTransitionPublicProps
>
export type CollapseTransitionInstance = InstanceType<DefineComponent<CollapseTransitionProps>>
