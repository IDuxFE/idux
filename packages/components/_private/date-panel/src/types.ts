/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { ExtractInnerPropTypes, ExtractPublicPropTypes, MaybeArray } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes, PropType } from 'vue'

export const datePanelProps = {
  cellTooltip: Function as PropType<(cell: { value: Date; disabled: boolean }) => string | void>,
  disabledDate: Function as PropType<(date: Date) => boolean>,
  type: {
    type: String as PropType<DatePanelType>,
    default: 'date',
  },
  value: [Date, Array] as PropType<Date | (Date | undefined)[]>,
  activeDate: Date,
  visible: {
    type: Boolean,
    default: undefined,
  },
  isSelecting: {
    type: Boolean,
    default: undefined,
  },

  onCellClick: [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
  onCellMouseenter: [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
  'onUpdate:activeDate': [Function, Array] as PropType<MaybeArray<(date: Date) => void>>,
}

export type DatePanelProps = ExtractInnerPropTypes<typeof datePanelProps>
export type DatePanelPublicProps = ExtractPublicPropTypes<typeof datePanelProps>
export type DatePanelComponent = DefineComponent<
  Omit<HTMLAttributes, keyof DatePanelPublicProps> & DatePanelPublicProps
>
export type DatePanelInstance = InstanceType<DefineComponent<DatePanelProps>>

export type DatePanelType = 'date' | 'week' | 'month' | 'quarter' | 'year'

// private
export const panelRowProps = {
  rowIndex: {
    type: Number,
    required: true,
  },
}

export const panelCellProps = {
  rowIndex: {
    type: Number,
    required: true,
  },
  cellIndex: {
    type: Number,
    required: true,
  },
}
