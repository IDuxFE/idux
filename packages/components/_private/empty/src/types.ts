/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { EmptyProps as IxEmptyProps } from '@idux/components/empty'

export interface EmptyProps {
  empty?: string | IxEmptyProps
}
