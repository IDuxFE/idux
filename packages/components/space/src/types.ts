/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { IxInnerPropTypes, IxPublicPropTypes } from '@idux/cdk/utils'
import type { DefineComponent, HTMLAttributes } from 'vue'

import { IxPropTypes } from '@idux/cdk/utils'

export type SpaceAlign = 'start' | 'center' | 'end' | 'baseline' | 'stretch'
export type SpaceJustify = 'start' | 'center' | 'end' | 'space-around' | 'space-between'
export type SpaceDirection = 'vertical' | 'horizontal'
export type SpaceSize = 'sm' | 'md' | 'lg'

export const spaceProps = {
  align: IxPropTypes.oneOf<SpaceAlign>(['start', 'center', 'end', 'baseline', 'stretch']).def('center'),
  block: IxPropTypes.bool,
  direction: IxPropTypes.oneOf<SpaceDirection>(['vertical', 'horizontal']),
  justify: IxPropTypes.oneOf<SpaceJustify>(['start', 'center', 'end', 'space-around', 'space-between']).def('start'),
  size: IxPropTypes.oneOfType([Number, String, IxPropTypes.array<string | number>()]),
  split: IxPropTypes.string,
  separator: IxPropTypes.string,
  vertical: IxPropTypes.bool,
  wrap: IxPropTypes.bool,
}

export type SpaceProps = IxInnerPropTypes<typeof spaceProps>
export type SpacePublicProps = IxPublicPropTypes<typeof spaceProps>
export type SpaceComponent = DefineComponent<Omit<HTMLAttributes, keyof SpacePublicProps> & SpacePublicProps>
export type SpaceInstance = InstanceType<DefineComponent<SpaceProps>>
