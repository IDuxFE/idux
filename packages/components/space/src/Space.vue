<template>
  <div class="ix-space" :class="className">
    <template v-for="(child, index) in spaceList" :key="child.id">
      <div class="ix-space-item" :class="child.className" :style="child.style">
        <component :is="child.node" />
      </div>
      <slot v-if="index < spaceList.length - 1" name="split">
        <span v-if="split" class="ix-space-split">
          {{ split }}
        </span>
      </slot>
    </template>
  </div>
</template>

<script lang="ts">
import type { ComputedRef, SetupContext, VNode } from 'vue'
import type { SpaceProps } from './types'

import { computed, defineComponent } from 'vue'
import { getSlotNodes, PropTypes } from '@idux/cdk/utils'
import { Logger } from '@idux/components/core/logger'
import { useGlobalConfig, SpaceConfig } from '@idux/components/core/config'

interface SpaceItem {
  style?: Partial<CSSStyleDeclaration>
  className?: string
  id: number
  node: VNode
}

const spaceSizePropType = PropTypes.oneOf(['small', 'medium', 'large'] as const)
export default defineComponent({
  name: 'IxSpace',
  props: {
    align: PropTypes.oneOf(['start', 'center', 'end', 'baseline'] as const).def('baseline'),
    direction: PropTypes.oneOf(['vertical', 'horizontal'] as const).def('horizontal'),
    size: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.oneOfType([spaceSizePropType, PropTypes.number])),
      PropTypes.oneOfType([spaceSizePropType, PropTypes.number]),
    ]),
    split: PropTypes.string,
    wrap: PropTypes.bool,
  },
  setup(props: SpaceProps, { slots }: SetupContext) {
    const spaceConfig = useGlobalConfig('space')
    const children = getSlotNodes(slots)
    const withSlotSplit = computed(() => !!slots.split)
    const spaceList = useSpaceList(props, spaceConfig, children, withSlotSplit)

    const className = useClassName(props)

    return { spaceList, className }
  },
})

function useSpaceList(
  props: SpaceProps,
  spaceConfig: SpaceConfig,
  children: VNode[],
  withSlotSplit: ComputedRef<boolean>,
) {
  return computed(() => {
    const size = props.size ?? spaceConfig.size
    const split = !!props.split || withSlotSplit.value

    let spaceList: SpaceItem[]
    if (Array.isArray(size)) {
      if (children.length - 1 !== size.length) {
        Logger.warn('The number of split elements is inconsistent with the length of the size array')
      }
      spaceList = children.map((child, index) => {
        const current: SpaceItem = { node: child, id: index }
        if (split) return current
        else if (typeof size[index] === 'number') {
          current.style = { marginRight: `${size[index]}px` }
        } else current.className = `ix-space-item-${size[index]}`
        return current
      })
    } else {
      spaceList = children.map((child, index) => {
        const current: SpaceItem = { node: child, id: index }
        if (split) return current
        else if (typeof size === 'number') {
          current.style = { marginRight: `${size}px` }
        } else current.className = `ix-space-item-${size}`
        return current
      })
    }
    return spaceList
  })
}

function useClassName(props: SpaceProps) {
  return computed(() => {
    const align = props.align
    const direction = props.direction
    const wrap = props.wrap

    return [`ix-space-${align}`, `ix-space-${direction}`, { 'ix-space-wrap': wrap }]
  })
}
</script>
