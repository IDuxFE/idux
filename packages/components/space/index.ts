/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import type { SpaceComponent } from './src/types'

import Space from './src/Space'

const IxSpace = Space as unknown as SpaceComponent

export { IxSpace }

export type {
  SpaceInstance,
  SpaceComponent,
  SpacePublicProps as SpaceProps,
  SpaceAlign,
  SpaceDirection,
  SpaceSize,
} from './src/types'

export { getThemeTokens as getSpaceThemeTokens } from './theme'

export type { SpaceThemeTokens } from './theme'
