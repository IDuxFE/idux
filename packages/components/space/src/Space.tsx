/**
 * @license
 *
 * Use of this source code is governed by an MIT-style license that can be
 * found in the LICENSE file at https://github.com/IDuxFE/idux/blob/main/LICENSE
 */

import { type ComputedRef, type VNode, computed, defineComponent, normalizeClass } from 'vue'

import { supportsFlexGap } from '@idux/cdk/platform'
import { Logger, convertCssPixel, flattenNode } from '@idux/cdk/utils'
import { useGlobalConfig } from '@idux/components/config'
import { type FormSize } from '@idux/components/form'
import { covertLabelVNode } from '@idux/components/utils'

import { spaceProps } from './types'

const flexGapSupported = supportsFlexGap()

const defaultSizeMap = {
  sm: '8px',
  md: '16px',
  lg: '24px',
} as const

export default defineComponent({
  name: 'IxSpace',
  props: spaceProps,
  setup(props, { slots }) {
    const common = useGlobalConfig('common')
    const mergedPrefixCls = computed(() => `${common.prefixCls}-space`)
    const config = useGlobalConfig('space')

    const wrap = computed(() => props.wrap ?? config.wrap)
    const vertical = computed(() => {
      const { direction, vertical } = props
      if (direction) {
        __DEV__ && Logger.warn('components/space', '`direction` was deprecated, please use `vertical` instead')
        return direction === 'vertical'
      }
      return vertical
    })
    const mergedGaps = computed(() => {
      const { size = config.size } = props
      const sizes = Array.isArray(size) ? size : [size, size]
      return sizes.map(size => defaultSizeMap[size as FormSize] || convertCssPixel(size))
    })

    const classes = computed(() => {
      const { align, justify, block } = props
      const prefixCls = mergedPrefixCls.value
      return normalizeClass({
        [prefixCls]: true,
        [`${prefixCls}-align-${align}`]: true,
        [`${prefixCls}-justify-${justify}`]: true,
        [`${prefixCls}-block`]: block,
        [`${prefixCls}-vertical`]: vertical.value,
        [`${prefixCls}-nowrap`]: !wrap.value,
      })
    })

    const style = computed(() => {
      const [rowGap, columnGap] = mergedGaps.value
      if (flexGapSupported) {
        return `gap: ${rowGap} ${columnGap}`
      } else {
        return !vertical.value && wrap.value ? `margin-bottom: -${convertCssPixel(rowGap)}` : undefined
      }
    })

    return () => {
      const nodes = flattenNode(slots.default?.())
      if (nodes.length === 0) {
        return
      }

      const prefixCls = mergedPrefixCls.value
      const children: VNode[] = []

      let separatorNode = covertLabelVNode(slots, props, 'split')
      if (separatorNode) {
        __DEV__ && Logger.warn('components/space', '`split` was deprecated, please use `separator` instead')
      } else {
        separatorNode = covertLabelVNode(slots, props, 'separator')
      }

      const lastIndex = nodes.length - 1
      nodes.forEach((node, index) => {
        const style = calcItemStyle(mergedGaps, wrap, vertical, index, lastIndex)
        children.push(
          <div key={`item-${index}`} class={`${prefixCls}-item`} style={style}>
            {node}
          </div>,
        )
        if (separatorNode && index < lastIndex) {
          children.push(
            <div key={`separator-${index}`} class={`${prefixCls}-item-separator`} style={style}>
              {separatorNode}
            </div>,
          )
        }
      })
      return (
        <div class={classes.value} style={style.value}>
          {children}
        </div>
      )
    }
  },
})

const calcItemStyle = (
  mergedGaps: ComputedRef<(string | number)[]>,
  wrap: ComputedRef<boolean>,
  vertical: ComputedRef<boolean | undefined>,
  index: number,
  lastIndex: number,
) => {
  if (flexGapSupported) {
    return undefined
  }

  const [rowGap, columnGap] = mergedGaps.value

  if (vertical.value) {
    const marginBottom = index < lastIndex ? convertCssPixel(rowGap) : undefined
    return { marginBottom }
  } else {
    const marginRight = index < lastIndex ? convertCssPixel(columnGap) : undefined
    const paddingBottom = wrap.value ? convertCssPixel(rowGap) : undefined
    return { marginRight, paddingBottom }
  }
}
