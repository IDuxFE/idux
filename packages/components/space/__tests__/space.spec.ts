import { MountingOptions, mount } from '@vue/test-utils'
import { createTextVNode, h } from 'vue'

import { renderWork } from '@tests'

import { IxButton } from '@idux/components/button'

import Space from '../src/Space'
import { SpaceProps } from '../src/types'

const defaultSlots = () => {
  return [createTextVNode('Space'), h(IxButton, { mode: 'primary' }, () => 'Button'), h(IxButton, {}, () => 'Button')]
}

describe('Space', () => {
  const SpaceMount = (options?: MountingOptions<Partial<SpaceProps>>) => {
    const { slots, ...rest } = options || {}
    return mount(Space, {
      ...rest,
      slots: { default: defaultSlots, ...slots },
    })
  }

  renderWork<SpaceProps>(Space, { slots: { default: defaultSlots } })

  test('align work', async () => {
    const wrapper = SpaceMount({ props: { align: 'start' } })

    expect(wrapper.classes()).toContain('ix-space-align-start')

    await wrapper.setProps({ align: 'end' })

    expect(wrapper.classes()).toContain('ix-space-align-end')
  })

  test('block work', async () => {
    const wrapper = SpaceMount({ props: { block: true } })

    expect(wrapper.classes()).toContain('ix-space-block')

    await wrapper.setProps({ block: false })

    expect(wrapper.classes()).not.toContain('ix-space-block')
  })

  test('justify work', async () => {
    const wrapper = SpaceMount({ props: { justify: 'start' } })

    expect(wrapper.classes()).toContain('ix-space-justify-start')

    await wrapper.setProps({ justify: 'end' })

    expect(wrapper.classes()).toContain('ix-space-justify-end')
  })

  test('size work', async () => {
    const wrapper = SpaceMount({ props: { size: 'sm' } })

    const wrapperElement = wrapper.element as HTMLElement
    const itemElements = wrapper.findAll('.ix-space-item').map(item => item.element as HTMLElement)

    expect(wrapperElement.style.marginBottom).toBe('-8px')
    expect(itemElements[0].style.marginRight).toBe('8px')
    expect(itemElements[0].style.paddingBottom).toEqual('8px')
    expect(itemElements[1].style.marginRight).toBe('8px')
    expect(itemElements[1].style.paddingBottom).toEqual('8px')
    expect(itemElements[2].style.marginRight).toBe('')
    expect(itemElements[2].style.paddingBottom).toEqual('8px')

    await wrapper.setProps({ size: 'md' })

    expect(wrapperElement.style.marginBottom).toBe('-16px')
    expect(itemElements[0].style.marginRight).toBe('16px')
    expect(itemElements[0].style.paddingBottom).toEqual('16px')
    expect(itemElements[1].style.marginRight).toBe('16px')
    expect(itemElements[1].style.paddingBottom).toEqual('16px')
    expect(itemElements[2].style.marginRight).toBe('')
    expect(itemElements[2].style.paddingBottom).toEqual('16px')

    await wrapper.setProps({ size: 30 })

    expect(wrapperElement.style.marginBottom).toBe('-30px')
    expect(itemElements[0].style.marginRight).toBe('30px')
    expect(itemElements[0].style.paddingBottom).toEqual('30px')
    expect(itemElements[1].style.marginRight).toBe('30px')
    expect(itemElements[1].style.paddingBottom).toEqual('30px')
    expect(itemElements[2].style.marginRight).toBe('')
    expect(itemElements[2].style.paddingBottom).toEqual('30px')

    await wrapper.setProps({ size: [16, 8] })

    expect(wrapperElement.style.marginBottom).toBe('-16px')
    expect(itemElements[0].style.marginRight).toBe('8px')
    expect(itemElements[0].style.paddingBottom).toEqual('16px')
    expect(itemElements[1].style.marginRight).toBe('8px')
    expect(itemElements[1].style.paddingBottom).toEqual('16px')
    expect(itemElements[2].style.marginRight).toBe('')
    expect(itemElements[2].style.paddingBottom).toEqual('16px')
  })

  test('separator work', async () => {
    const wrapper = SpaceMount({ props: { separator: '/' } })

    expect(wrapper.findAll('.ix-space-item-separator').length).toBe(2)
    expect(wrapper.find('.ix-space-item-separator').text()).toBe('/')

    await wrapper.setProps({ separator: '-' })

    expect(wrapper.find('.ix-space-item-separator').text()).toBe('-')
  })

  test('vertical work', async () => {
    const wrapper = SpaceMount({ props: { vertical: true } })

    expect(wrapper.classes()).toContain('ix-space-vertical')

    await wrapper.setProps({ vertical: false })

    expect(wrapper.classes()).not.toContain('ix-space-vertical')
  })

  test('wrap work', async () => {
    const wrapper = SpaceMount({ props: { wrap: false } })

    expect(wrapper.classes()).toContain('ix-space-nowrap')

    await wrapper.setProps({ wrap: true })

    expect(wrapper.classes()).not.toContain('ix-space-nowrap')
  })
})
