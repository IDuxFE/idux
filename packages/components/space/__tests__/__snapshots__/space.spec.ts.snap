// Vitest Snapshot v1

exports[`Space > render work 1`] = `
"<div class=\\"ix-space\\" style=\\"margin-bottom: -8px;\\">
  <div class=\\"ix-space-item\\" style=\\"margin-right: 8px; padding-bottom: 8px;\\">Space</div>
  <div class=\\"ix-space-item\\" style=\\"margin-right: 8px; padding-bottom: 8px;\\"><button class=\\"ix-button ix-button-primary ix-button-md\\" type=\\"button\\"><span>Button</span>
      <div aria-hidden=\\"true\\" class=\\"ix-wave\\"></div>
    </button></div>
  <div class=\\"ix-space-item\\" style=\\"padding-bottom: 8px;\\"><button class=\\"ix-button ix-button-default ix-button-md\\" type=\\"button\\"><span>Button</span>
      <div aria-hidden=\\"true\\" class=\\"ix-wave\\"></div>
    </button></div>
</div>"
`;
